src/err.rs

```diff
@@ -56,7 +57,21 @@ pub struct PyErr {
 pub type PyResult<T> = Result<T, PyErr>;
 
 /// Marker type that indicates an error while downcasting
-pub struct PyDowncastError;
+#[derive(Debug)]
+pub struct PyDowncastError<'a, 'b> {
+    from: &'a PyAny,
+    to: Cow<'b, str>,
```

Comment:
Isn't this too complex?
I prefer to `Cow<'static str>` or `String`.

Reply:
I think we should just be able to do `Cow<'a, str>` rather than needing a separate lifetime? Agreed having two lifetimes is too much. I'd rather avoid the copy into a `String` if possible!

Maybe you're right and non-copy case is always `'static`, so perhaps `Cow<'static, str>' would work. Or yeah just `&'static str` if there's now no uses where we create a new `String`.

Reply:
A `String` is allocated anyways when the conversion to a `PyErr` happens. The lifetime of `to` is generally not tied to the object that we try to convert from.

If you don't like the `'b` parameter, I think @kngwyu's suggestion of setting it to `'static` is the easiest. Otherwise, I don't think the additional `String` allocation would hurt, if it was set to be a `String`.

Reply:
My motivation for avoiding the `String` allocation is that then users who are handling this error on the Rust side (e.g. never converting to a `TypeError`) can have as little overhead as possible. A small thing but they'll add up over time!

I'm happy with `Cow<'static, str>` here 👍

Reply:
Thanks for pointing out! I didn't see that `PyTryFrom` returns `PyDowncastError` instead of `PyErr`.